src/System.Collections/src/System/Collections/Generic/SortedSet.TreeSubSet.cs

```diff
@@ -0,0 +1,358 @@
+                    current = stack.Pop();
+                    if (!action(current))
+                    {
+                        return false;
+                    }
+
+                    Node node = (reverse ? current.Left : current.Right);
+                    while (node != null)
+                    {
+                        if (IsWithinRange(node.Item))
+                        {
+                            stack.Push(node);
+                            node = (reverse ? node.Right : node.Left);
+                        }
+                        else if (_lBoundActive && Comparer.Compare(_min, node.Item) > 0)
+                        {
+                            node = node.Right;
+                        }
+                        else
+                        {
+                            node = node.Left;
+                        }
+                    }
+                }
+                return true;
+            }
+
+            internal override bool BreadthFirstTreeWalk(TreeWalkPredicate<T> action)
+            {
+                VersionCheck();
+
+                if (_root == null)
+                {
+                    return true;
+                }
+
+                Queue<Node> processQueue = new Queue<Node>();
+                processQueue.Enqueue(_root);
+                Node current;
+
+                while (processQueue.Count != 0)
+                {
+                    current = processQueue.Dequeue();
+                    if (IsWithinRange(current.Item) && !action(current))
+                    {
+                        return false;
+                    }
+                    if (current.Left != null && (!_lBoundActive || Comparer.Compare(_min, current.Item) < 0))
+                    {
+                        processQueue.Enqueue(current.Left);
+                    }
+                    if (current.Right != null && (!_uBoundActive || Comparer.Compare(_max, current.Item) > 0))
+                    {
+                        processQueue.Enqueue(current.Right);
+                    }
+                }
+                return true;
+            }
+
+            internal override SortedSet<T>.Node FindNode(T item)
+            {
+                if (!IsWithinRange(item))
+                {
+                    return null;
+                }
+
+                VersionCheck();
+#if DEBUG
+                Debug.Assert(this.versionUpToDate() && _root == _underlying.FindRange(_min, _max));
+#endif
+                return base.FindNode(item);
+            }
+
+            // this does indexing in an inefficient way compared to the actual sortedset, but it saves a
+            // lot of space
+            internal override int InternalIndexOf(T item)
+            {
+                int count = -1;
+                foreach (T i in this)
+                {
+                    count++;
+                    if (Comparer.Compare(item, i) == 0)
+                        return count;
+                }
+#if DEBUG
```

Comment:
Parameter evaluation is omitted, even if it can have side effects. A quick demonstration is:
```C#
int x;
Debug.Assert(int.TryParse("3", out x));
Console.WriteLine(x);
```
This won't compile if `DEBUG` isn't defined, because the only place `x` is assigned is left out of the compilation and hence CS0165.

Long story short, all the `#if DEBUG`s can be left out.